dotnet/samples/KernelSyntaxExamples/Example51_StepwisePlanner.cs

```diff
@@ -12,77 +15,213 @@
+                currentExecutionResult.answer = result.Result;
+            }
+
+            if (result.Variables.TryGetValue("stepCount", out string? stepCount))
+            {
+                Console.WriteLine("Steps Taken: " + stepCount);
+                currentExecutionResult.stepsTaken = stepCount;
+            }
+
+            if (result.Variables.TryGetValue("skillCount", out string? skillCount))
+            {
+                Console.WriteLine("Skills Used: " + skillCount);
+            }
+
+            if (result.Variables.TryGetValue("iterations", out string? iterations))
+            {
+                Console.WriteLine("Iterations: " + iterations);
+                currentExecutionResult.iterations = iterations;
+            }
+        }
+#pragma warning disable CA1031
+        catch (Exception ex)
         {
-            Console.WriteLine("Skills Used: " + skillCount);
+            Console.WriteLine("Exception: " + ex);
         }
 
         Console.WriteLine("Time Taken: " + sw.Elapsed);
+        currentExecutionResult.timeTaken = sw.Elapsed.ToString();
+        ExecutionResults.Add(currentExecutionResult);
         Console.WriteLine("*****************************************************");
     }
 
-    private static IKernel GetKernel()
+    private static IKernel GetKernel(ref ExecutionResult result, bool useChat = false, string? model = null)
     {
         var builder = new KernelBuilder();
+        if (useChat)
+        {
+            builder.WithAzureChatCompletionService(
+                model ?? ChatModelOverride ?? TestConfiguration.AzureOpenAI.ChatDeploymentName,
+                TestConfiguration.AzureOpenAI.Endpoint,
+                TestConfiguration.AzureOpenAI.ApiKey,
+                alsoAsTextCompletion: true,
+                setAsDefault: true);
+
+            Console.WriteLine($"Model: {model ?? ChatModelOverride ?? TestConfiguration.AzureOpenAI.ChatDeploymentName} ({ChatMaxTokens ?? (new Microsoft.SemanticKernel.Planning.Stepwise.StepwisePlannerConfig()).MaxTokens})");
+            result.model = model ?? ChatModelOverride ?? TestConfiguration.AzureOpenAI.ChatDeploymentName;
+        }
+        else
+        {
+            builder.WithAzureTextCompletionService(
+                model ?? TextModelOverride ?? TestConfiguration.AzureOpenAI.DeploymentName,
+                TestConfiguration.AzureOpenAI.Endpoint,
+                TestConfiguration.AzureOpenAI.ApiKey);
 
-        builder.WithAzureChatCompletionService(
-            TestConfiguration.AzureOpenAI.ChatDeploymentName,
-            TestConfiguration.AzureOpenAI.Endpoint,
-            TestConfiguration.AzureOpenAI.ApiKey,
-            alsoAsTextCompletion: true,
-            setAsDefault: true);
+            Console.WriteLine($"Model: {model ?? TextModelOverride ?? TestConfiguration.AzureOpenAI.DeploymentName} ({TextMaxTokens ?? (new Microsoft.SemanticKernel.Planning.Stepwise.StepwisePlannerConfig()).MaxTokens})");
```

Comment:
Improve formatting for better readability. It's important since example code should be easy to follow.